resource/storage.go

```diff
@@ -20,7 +20,7 @@ type Storer interface {
 	// If the fetching of the data is not immediate, the method must listen for cancellation
 	// on the passed ctx. If the operation is stopped due to context cancellation, the
 	// function must return the result of the ctx.Err() method.
-	Find(ctx context.Context, lookup *Lookup, page, perPage int) (*ItemList, error)
+	Find(ctx context.Context, lookup *Lookup, page, perPage int, skip int) (*ItemList, error)
```

Comment:
BTW, why naming this new param `skip` instead of `offset` as it is named in the query-string and in Mongo?